resident/resident-service/src/main/java/io/mosip/resident/service/impl/ProxyMasterdataServiceImpl.java

```diff
@@ -413,5 +462,118 @@ public ResponseWrapper<?> getGenderTypesByLangCode(String langCode) throws Resid
+		logger.debug("ResidentServiceImpl::getResidentServicePDF()::entry");
+		ResponseWrapper<?> proxyResponseWrapper = proxyMasterdataService
+				.getAllTemplateBylangCodeAndTemplateTypeCode(langCode, REGISTRATION_CENTER_TEMPLATE_NAME);		
+		ResponseWrapper<?> regCentResponseWrapper = proxyMasterdataService.getRegistrationCentersByHierarchyLevel(langCode,hierarchyLevel, name);
+		Map<String,Object> regCentersMap = new LinkedHashMap<>();
+		List<RegistrationCenterDto >regCentersDtlsList=Collections.emptyList();		
+		if (regCentResponseWrapper != null) {
+			RegistrationCenterResponseDto registrationCentersDtls = mapper.readValue(
+					mapper.writeValueAsString(regCentResponseWrapper.getResponse()), RegistrationCenterResponseDto.class);
+			List<RegistrationCenterDto> regCenterIntialList = registrationCentersDtls.getRegistrationCenters();
+			if(regCenterIntialList!=null && !regCenterIntialList.isEmpty()) {
+				IntStream.range(0, regCenterIntialList.size()).forEach(i-> {
+					try {
+						addRegistrationCenterDtls(i, regCenterIntialList.get(i));
+					} catch (Exception e) {
+						throw new ResidentServiceException(ResidentErrorCode.UNABLE_TO_PROCESS,e);
+					}
+				});
+			}
+			regCentersMap.put("regCenterIntialList", regCenterIntialList);
+		}
+		logger.debug("template data from DB:" + proxyResponseWrapper.getResponse());
+		Map<String, Object> templateResponse = new LinkedHashMap<>((Map<String, Object>) proxyResponseWrapper.getResponse());
+		String fileText = (String) templateResponse.get("fileText");
+		InputStream downLoadRegCenterTemplate = new ByteArrayInputStream(fileText.getBytes(StandardCharsets.UTF_8));
+		InputStream downLoadRegCenterTemplateData = templateManager.merge(downLoadRegCenterTemplate, regCentersMap);
+		
+		StringWriter writer = new StringWriter();
+		IOUtils.copy(downLoadRegCenterTemplateData, writer, "UTF-8");
+		ByteArrayOutputStream pdfValue = (ByteArrayOutputStream) pdfGenerator.generate(writer.toString());
+		logger.debug("ResidentServiceImpl::residentServiceHistoryPDF()::exit");
+		return convertOutputStreamToInputStream(pdfValue);
+	}
+	
+	/**
+	 * update the registrationcenter details
+	 */
+	private void addRegistrationCenterDtls(int index, RegistrationCenterDto regCenterDto)
+			throws ResidentServiceCheckedException, Exception {
+		String workingHours = "";
+		String fullAddress = getFullAddress(regCenterDto.getAddressLine1(), regCenterDto.getAddressLine2(),
+				regCenterDto.getAddressLine3());
+		regCenterDto.setSerialNumber(index+1);
+		regCenterDto.setFullAddress(fullAddress);
+		List<WorkingDaysDto>  workingDaysList = getRegCenterWorkingDays(regCenterDto.getId(), regCenterDto.getLangCode());
+		workingHours = workingDaysList.get(0).getName() + "-" + workingDaysList.get(1).getName() + "|"
+				+ regCenterDto.getCenterStartTime().substring(0, 5) + " " + "am" + "-"
+				+ regCenterDto.getCenterEndTime().substring(0, 5) + " " + "pm";
```

Comment:
same as earlier do not hard code am or pm it should be derived.

Reply:
Modified the code as per the review comments. View code changes in DownLoadMasterDataServiceImpl.java